sdk/python/kfp_tekton/compiler/_op_to_template.py

```diff
@@ -505,4 +532,59 @@ def _op_to_template(op: BaseOp,
             template.setdefault('metadata', {}).setdefault('annotations', {})['pipelines.kubeflow.org/max_cache_staleness'] = \
                 str(op.execution_options.caching_strategy.max_cache_staleness)
 
+    # Sort and arrange results based on provided estimate size and process results in multi-steps if the result sizes are too big.
+    result_size_map = "{}"
+    if processed_op.pod_annotations:
+        result_size_map = processed_op.pod_annotations.get("tekton-result-sizes", "{}")
+    # Only sort and arrange results when the estimated sizes are given.
+    if result_size_map and result_size_map != "{}":
+        try:
+            result_size_map = json.loads(result_size_map)
+        except ValueError:
+            raise("tekton-result-sizes annotation is not a valid JSON")
+        # Sort key orders based on values
+        result_size_map = dict(sorted(result_size_map.items(), key=lambda item: item[1]))
+        max_byte_size = 2048
+        verified_result_size_map = {0: {}}
+        op_result_names = [name['name'] for name in template['spec']['results']]
+        step_bins = {0: 0}
+        step_counter = 0
+        # Group result files to not exceed max_byte_size
+        # TODO: group result files with better bin packing algorithm
```

Comment:
I don't think it matters that much, but as for a better algorithm, how about the following:
1. sort the results from biggest to smallest
2. while some results are left to process:
   1. find the biggest result that still fits into the current step
   2. if no result fits, start a new segment
   3. if some result fits, put it into the current segment
3. reverse the steps' order

That could handle situations like:
* 100
* 200
* 400
* 1500
* 1600

currently:

| step | result |
|:----:|-------:|
|      |        |
|   2  |   1600 |
|      |        |
|   1  |   1500 |
|      |        |
|   0  |    400 |
|   0  |    200 |
|   0  |    100 |


new:

| step | result |
|:----:|-------:|
|      |        |
|   1  |   1600 |
|   1  |    400 |
|      |        |
|   0  |   1500 |
|   0  |    200 |
|   0  |    100 |

Reply:
Updated the algorithm to add results from large to small, should be optimized as above.